Activate node sgtk

j0yu opened the pull request:
For Release `v1.6.1+wwfx.1.1.0`

### Added

- `NodeHandlerBase.activate_sgtk()` for `tk-multi-loader2` hooks.
- `python/tk_houdini/base_hooks/node_handler_base.py` module docstrings
- `HoudiniEngine.apply_external_stylesheet(widget)` which nicely expose the parent class's `_apply_external_stylesheet`

### Fixed

- Warns if category/node type is invalid instead of raising error
- `int(current)` evaluated too early when it can be a version policy string

### Changed

- Trimmed `######` separators

-----

~~I got 2 ways~~ for dealing with the following repetition in `{config}/hooks/tk-multi-loader2/tk-houdini_actions.py`

```python
        if node_handler:
            node_handler.populate_node_from_publish_data(archive_node, sg_publish_data)
            use_sgtk = archive_node.param("use_sgtk")
            if not use_sgtk.eval():
                use_sgtk.set(True)
                node_handler._enable_sgtk(archive_node, True)
```

**Options:**

1. ~~`populate_activate_sgtk()` in `hooks/node_handlers/base_import_handler.py`~~ <details><summary> Using option 2 now instead of this </summary>

    ```python
        if node_handler:
            node_handler.populate_activate_sgtk(archive_node, sg_publish_data)
    ```
    </details>

2. More granular `activate_sgtk()` in `python/tk_houdini/base_hooks/node_handler_base.py`, which would be 1 line extra but more extendable as it lives in the base hook

    ```python
        if node_handler:
            node_handler.populate_node_from_publish_data(archive_node, sg_publish_data)
            node_handler.activate_sgtk(archive_node)
    ```

Commit:
Stripped extra long separators, added docs

Commit:
Added populate_activate_sgtk

Commit:
Alternative activate_sgtk

Commit:
Warns if category/node type is invalid

Liametc commented:
I am inclined to go with option 2, incase we need to do those two functionalities separately at some point in the future

Commit:
parm not param

Commit:
Goodbye option 1

Commit:
Little bit more docstrings to activate_sgtk

j0yu commented:
Should be good review now

Commit:
Fix for premature int(current)

current can be a version policy string

Commit:
Fixed the fix

Commit:
Nicer apply_external_stylesheet

j0yu commented:
I fixed the fix and added [HoudiniEngine.apply_external_stylesheet()](https://github.com/wwfxuk/tk-houdini/pull/2/files#diff-ae7891bac0990fc35610210916ecd032R678-R683)
